### PR TITLE
chore: bump dev dependencies

### PR DESCRIPTION
**Pull request by scagood**
1. Bumped all packages (except eslint) to the latest version.
1. Replaced `eslint-plugin-node` with `eslint-plugin-n`

(#471)

### CODE REVIEW

**Comment by scagood** [comment]
All linting errors are because of #482

**Comment by scagood** [comment]
🤔 So now comes the time I need to know if we:
- keep supporting eslint v8 (Roll all versions back to points where [12.22](https://github.com/eslint/eslint/blob/v8.57.0/package.json#L177))
- Drop support for eslint v8 (Bump package engines to match eslint v9 [^18.18.0 || ^20.9.0 || >=21.1.0](https://github.com/eslint/eslint/blob/v9.0.0/package.json#L179)) (eg in [fc861bf](https://github.com/eslint-community/eslint-plugin-promise/commit/fc861bf231d73d5fbd065946c922ac89f27f4f66))

**Comment by brettz9** [comment]
Can't we keep supporting ESLint v8 while bumping the Node requirements? We don't have to support every possible install which can handle ESLint 8.

**Comment by scagood** [comment]
> Can't we keep supporting ESLint v8 while bumping the Node requirements? We don't have to support every possible install which can handle ESLint 8.

Ah, yes, sorry I miss wrote, you're 100% right, it'd be just dropping node versions for the eslint version! 😅

**Comment by brettz9** [comment]
I think the bumped Node may also address your CI issues.

**Comment by scagood** [comment]
> I think the bumped Node may also address your CI issues.

You'd be correct there :)

File: package.json [hunk 1]
```diff
@@ -66,22 +66,22 @@
...
+    "eslint-plugin-jest": "^28.5.0",
+    "eslint-plugin-n": "^17.7.0",
+    "eslint-plugin-prettier": "^5.1.3",
+    "globals": "^15.3.0",
+    "husky": "^9.0.11",
+    "jest": "^29.7.0",
+    "lint-staged": "^15.2.5",
+    "npm-run-all2": "^6.2.0",
+    "prettier": "^3.2.5",
+    "typescript": "^5.4.5"
```
Comment (voxpelli) [inline package.json:84]:
```suggestion
    "typescript": "~5.4.5"
```

Typescript can be breaking in minor versions

**Review by voxpelli — APPROVED** [review]

**Review by voxpelli — APPROVED** [review]

**Review by scagood — COMMENTED** [review]
Its about now, I realised I missed actually posting these comments :eyes:

File: package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "lint": "npm-run-all \"lint:*\"",
     "lint:eslint-docs": "npm run update:eslint-docs && git diff --exit-code",
     "lint:js": "eslint --report-unused-disable-directives .",
-    "prepare": "husky install",
+    "prepare": "husky",
```
[the remark this answers is not in the document]
Comment (voxpelli) [reply package.json:29]: It at least brings convention, I like it sometimes

File: README.md [hunk 3]
```diff
@@ -96,7 +96,8 @@ or start with the recommended rule set:
 🚫 Configurations disabled in.\
 ✅ Set in the `flat/recommended` configuration.\
 ✅ Set in the `recommended` configuration.\
-🔧 Automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--fix).
+🔧 Automatically fixable by the
+[`--fix` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--fix).
```
[the remark this answers is not in the document]
Comment (voxpelli) [reply README.md:100]: Ugh, I would like to remove prettier :/

**Pull request merged**